src/backend/src/services/teams.services.ts

```diff
@@ -162,4 +162,59 @@ export default class TeamsService {
 
     return teamTransformer(updateTeam);
   }
+
+  /**
+   * Update the given teamId's team's leads
+   * @param submitter a user who's making this request
+   * @param teamId a id of team to be updated
+   * @param userIds a array of user Ids that replaces team's old leads
+   * @returns an updated team
+   * @throws if the team is not found, the submitter has no priviledge, or any user from the given userIds does not exist
+   */
+  static async setTeamLeads(submitter: User, teamId: string, userIds: number[]): Promise<Team> {
+    const team = await prisma.team.findUnique({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    const teams = await prisma.team.findMany();
+
+    const newLeads = await getUsers(userIds);
+
+    if (!team) throw new NotFoundException('Team', teamId);
+
+    if (!isAdmin(submitter.role) && submitter.userId !== team.headId) {
+      throw new AccessDeniedException('You must be an admin or the head to update the lead!');
+    }
+
+    if (newLeads.map((lead) => lead.userId).includes(team.headId)) {
```

Comment:
these bottom 3 should be HttpExceptions with the 400 error code, we should also add tests for each of these

Reply:
To add on, we should also add a test that validates if none of the new leads are a lead on another team.

Reply:
Can we also make a separate variable for `newLeads.map((lead) => lead.userId)` into newLeadIds? This variable will be used multiple times for validating any HttpExceptions, and using it will make us read the code easier, especially on line 198 where we have to validate if Leads is not a head or a lead on another existing team.

Reply:
I dont think we should limit leads to one team